### DISCUSSION TITLE
How to handle exception RecordNotFound

### DISCUSSION THREAD

**Discussion by devalnor** [prisma/prisma#4552] [Q&A] [answered]
How that handle RecordNotFound exception from other exceptions

When we delete a record that doesn't exist, prisma return `P2016` error with details:
`Error for binding '0': RecordNotFound("Record to delete does not exist.")`

Ok but P2016 seems to be too large for any kind of query error.
`P2016 = "Query interpretation error. ${details}"`

I can  check if `error.detail` contain the keyword `RecordNotFound` but the this is a dirty hack.

Documentation say:
> delete returns a plain old JavaScript object or throws a **RecordNotFound** exception if the unique identifier does not exist:
https://www.prisma.io/docs/concepts/components/prisma-client/crud#delete

Any suggestion or idea ?

**Comment by ryands17** [comment, marked as answer]
@devalnor 👋 
The error `RecordNotFound` is actually thrown by the query engine and is not a native JavaScript error object. Unfortunately the only way to validated this would be to check for `error.detail`.

**Reply by AlbertLacambraNexplore** [reply]
it would be good to get proper errors

**Reply by ryands17** [reply]
Currently you could filter errors via the [error code](https://www.prisma.io/docs/reference/api-reference/error-reference) and that would make it a bit better in this case. This can be obtained via `err.code` where `err` is the error object.

**Reply by janpio** [reply]
Please someone open an issue about this - we want to throw a much better, handleable error here.

**Reply by JacobMuchow** [reply]
Done! I encountered this today as well.

https://github.com/prisma/prisma/issues/8198

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
